purchase_deposit/wizard/purchase_make_invoice_advance.py

```diff
@@ -163,10 +163,10 @@ def create_invoices(self):
         product = self.purchase_deposit_product_id
         if not product:
             vals = self._prepare_deposit_product()
-            product = self.purchase_deposit_product_id = self.env[
-                "product.product"
-            ].create(vals)
-            self.env.company.purchase_deposit_product_id = product
+            product = self.purchase_deposit_product_id = (
+                self.env["product.product"].sudo().create(vals)
+            )
+            self.sudo().env.company.purchase_deposit_product_id = product
```

Comment:
Maybe it doesn't really matter, but to limit the scope...
```suggestion
            self.env.company.sudo().purchase_deposit_product_id = product
```